Modules/socketmodule.c

```diff
@@ -6845,15 +6845,25 @@ static PyObject *
 socket_if_indextoname(PyObject *self, PyObject *arg)
 {
 #ifdef MS_WINDOWS
+    NET_IFINDEX index_long;
     NET_IFINDEX index;
 #else
-    unsigned long index;
+    unsigned long index_long;
+    unsigned int index;
 #endif
     char name[IF_NAMESIZE + 1];
 
-    index = PyLong_AsUnsignedLong(arg);
-    if (index == (unsigned long) -1)
+    index_long = PyLong_AsUnsignedLong(arg);
+    if (index_long == (unsigned long) -1 && PyErr_Occurred()) {
+        return NULL;
+    }
+
+    index = (unsigned int)index_long;
+
+    if ((unsigned long)index != index_long) {
```

Comment:
On Windows, NET_IFINDEX type is unsigned long, and so this change is not correct:
https://github.com/wine-mirror/wine/blob/11cb3bbf6908261458e8b6c0ad44f7daab8b9266/include/ifdef.h#L26

Reply:
This change does not have effect on Windows, but if fixes integer overflow on other platforms, where the argument is `unsigned int`.